### PR TITLE
wifi: Migrate to nRF70 upstream

### PR DESCRIPTION
**Pull request by krish2718**
The upstream PR for nRF70 support (Boards, Drivers and hostap) ~will soon be~ is merged and this PR removes nRF70 support from NCS and switches to use from Zephyr upstream.

For now only nRF7002DK and EK (including variants) are supported, other shields (54 series, EB, Thingy etc) will be added later. Also, Firmware patches in external flash isn't supported, this will also be added later.

The invasive changes which are treewide are minor ones as below:

1. rename configurations from NRF700X -> NRF70
2. rename configurations from WPA_SUPP -> WIFI_NM_WPA_SUPPLICANT
3. use per-module min HEAP https://docs.zephyrproject.org/latest/kernel/memory_management/heap.html#defining-the-heap-memory-pool this can introduce a CMake warning, so, to fix that had to modify samples to explicitly enable `CONFIG_HEAP_MEM_POOL_IGNORE_MIN=y`

### CODE REVIEW

File: doc/nrf/protocols/matter/getting_started/memory_optimization.rst
Comment (nordicjm) [inline doc/nrf/protocols/matter/getting_started/memory_optimization.rst:396]: table needs adjusting

File: sysbuild/Kconfig.wifi
Comment (nordicjm) [inline sysbuild/Kconfig.wifi:10]: Whilst I'm not against this change, this does just break existing customers for no real reason. It needs migration notes adding

File: sysbuild/Kconfig.wifi
Comment (nordicjm) [inline sysbuild/Kconfig.wifi:19]: Shouldn't the text remain as nRF700x? Or nRF70x? Since this doesn't apply to just a single product called nRF70 but the series

**Review by nordicjm — APPROVED** [review]

File: doc/nrf/releases_and_maturity/releases/release-notes-2.6.0.rst
Comment (trantanen) [inline doc/nrf/releases_and_maturity/releases/release-notes-2.6.0.rst:848]: I don't think we should rename old release notes. Normally I've just removed linking so doc build succeeds but tech writers can comment:
```suggestion
  * WPA supplicant advanced features in all Matter samples using the ``CONFIG_WPA_SUPP_ADVANCED_FEATURES`` Kconfig option.
```

File: sysbuild/Kconfig.wifi
Comment (krish2718) [reply sysbuild/Kconfig.wifi:10]: It was part of the cleanup suggested in upstream review, I will add it to migration notes.

File: sysbuild/Kconfig.wifi
Comment (krish2718) [reply sysbuild/Kconfig.wifi:19]: `the nRF70 Series` is appropriate.

File: samples/cellular/location/overlay-nrf700x-wifi-scan-only.conf
Comment (trantanen) [inline samples/cellular/location/overlay-nrf700x-wifi-scan-only.conf:69]: Is this also needed for modem_shell?
Or why it wouldn't be?
I'm trying to understand what the minimum value is that this flag wouldn't be needed but cannot find any documentation on that.

Comment (krish2718) [reply samples/cellular/location/overlay-nrf700x-wifi-scan-only.conf:69]: nRF70 driver sets it a higher values ~150K, so, if a sample uses less than that then this needs to be enabled else you will see a build warning + it will use driver's default causing memory overflows.

> Is this also needed for modem_shell?
Or why it wouldn't be?

Yes, missed that, will add.

File: doc/nrf/releases_and_maturity/releases/release-notes-2.6.0.rst
Comment (krish2718) [reply doc/nrf/releases_and_maturity/releases/release-notes-2.6.0.rst:848]: Ah, I just reverted this change, I will update to remove the linkage to Kconfig.

File: sysbuild/Kconfig.wifi
Comment (krish2718) [reply sysbuild/Kconfig.wifi:19]: I will do this in a separate PR upstream and then pull to NCS.

**Review by plskeggs — APPROVED** [review]
nRF Cloud sample changes are ok

**Review by glarsennordic — APPROVED** [review]

File: doc/nrf/releases_and_maturity/migration/migration_guide_2.8.rst
Comment (trantanen) [inline doc/nrf/releases_and_maturity/migration/migration_guide_2.8.rst:36]: I think we should clearly mention that customer should replace `CONFIG_WIFI_NRF700X*` Kconfig options with `CONFIG_WIFI_NRF700X*` Kconfig options. See below how clearly functions and Kconfig option is mentioned for lte_lc.

File: doc/nrf/releases_and_maturity/releases/release-notes-2.4.0.rst
Comment (krish2718) [reply doc/nrf/releases_and_maturity/releases/release-notes-2.4.0.rst:656]: @richabp

File: doc/nrf/releases_and_maturity/migration/migration_guide_2.8.rst
Comment (krish2718) [reply doc/nrf/releases_and_maturity/migration/migration_guide_2.8.rst:36]: Does it have to be an exhaustive list? They are too many :-(.

File: doc/nrf/releases_and_maturity/migration/migration_sysbuild.rst
Comment (richabp) [inline doc/nrf/releases_and_maturity/migration/migration_sysbuild.rst:271]:
```suggestion
|               ``SB_CONFIG_WIFI_NRF70_SCAN_ONLY``               | Use Scan-only mode firmware patches and set application to this mode        |
```

File: doc/nrf/releases_and_maturity/migration/migration_sysbuild.rst
Comment (richabp) [inline doc/nrf/releases_and_maturity/migration/migration_sysbuild.rst:273]:
```suggestion
|               ``SB_CONFIG_WIFI_NRF70_RADIO_TEST``              | Use Radio Test mode firmware patches and set application to this mode       |
```

File: doc/nrf/releases_and_maturity/migration/migration_sysbuild.rst
Comment (richabp) [inline doc/nrf/releases_and_maturity/migration/migration_sysbuild.rst:275]:
```suggestion
|               ``SB_CONFIG_WIFI_NRF70_SYSTEM_WITH_RAW_MODES``   | Use system with Raw modes firmware patches and set application to this mode |
```

File: doc/nrf/releases_and_maturity/migration/migration_guide_2.8.rst
Comment (richabp) [inline doc/nrf/releases_and_maturity/migration/migration_guide_2.8.rst:38]:
```suggestion
nRF70 Series
------------

```

File: doc/nrf/releases_and_maturity/migration/migration_guide_2.8.rst
Comment (richabp) [inline doc/nrf/releases_and_maturity/migration/migration_guide_2.8.rst:48]:
```suggestion
   * The nRF70 Series support is now part of Zephyr upstream, necessitating the following changes:

     * The nRF70 Series driver namespace has been renamed from ``NRF700X`` to ``NRF70``.
       Update your application to use the new namespace.
     * The nRF70 Series driver now uses a per-module kernel heap with a higher default.
      If a sample or an application uses the kernel heap less than the default size, a build warning is displayed.
      Use the :kconfig:option:`CONFIG_HEAP_MEM_POOL_IGNORE_MIN` Kconfig option and enable it to suppress the warning.
```

File: doc/nrf/releases_and_maturity/migration/migration_sysbuild.rst
Comment (richabp) [reply doc/nrf/releases_and_maturity/migration/migration_sysbuild.rst:275]: Can you please correct the mention of **nRF7000x** in line 307 which is incorrect?

**Review by richabp — CHANGES_REQUESTED** [review]

File: doc/nrf/releases_and_maturity/migration/migration_guide_2.8.rst
Comment (krish2718) [reply doc/nrf/releases_and_maturity/migration/migration_guide_2.8.rst:36]: discussed offline with @richabp we both felt it's better leave it to generic as there are too many Kconfigs and its a trivial namespace rename.

File: doc/nrf/releases_and_maturity/releases/release-notes-2.4.0.rst
Comment (krish2718) [reply doc/nrf/releases_and_maturity/releases/release-notes-2.4.0.rst]: After offline discussion, resolved to using pure text

File: doc/nrf/releases_and_maturity/migration/migration_guide_2.8.rst
Comment (krish2718) [reply doc/nrf/releases_and_maturity/migration/migration_guide_2.8.rst]: Resolving this but if you disagree feel free to unresolve and we can discuss.

File: doc/nrf/releases_and_maturity/releases/release-notes-2.6.0.rst
Comment (richabp) [reply doc/nrf/releases_and_maturity/releases/release-notes-2.6.0.rst]: Agree. Change it to ``CONFIG_WPA_SUPP_ADVANCED_FEATURES``.

Comment (krish2718) [reply doc/nrf/releases_and_maturity/releases/release-notes-2.6.0.rst]: Already did, thanks.

File: doc/nrf/releases_and_maturity/migration/migration_guide_2.8.rst
Comment (richabp) [inline doc/nrf/releases_and_maturity/migration/migration_guide_2.8.rst:67]:
```suggestion
    Use the :kconfig:option:`CONFIG_HEAP_MEM_POOL_IGNORE_MIN` Kconfig option and enable it to suppress the warning.
```

**Review by richabp — CHANGES_REQUESTED** [review]

File: doc/nrf/releases_and_maturity/migration/migration_guide_2.8.rst
Comment (trantanen) [reply doc/nrf/releases_and_maturity/migration/migration_guide_2.8.rst]: I wasn't looking for a big list but a version with `*` character which I wrote in my original comment.
I'm trying to make migrations easier as we see complaints that migrations from one release to another are painful but we still continue to cause more and more changes during the migrations. This is a trivial namespace change but if we can save 5mins for one unexperienced developer, it's worth it (and I honestly believe we save a lot more no matter how trivial this feels).
But up to you and tech writers.

Comment (krish2718) [reply doc/nrf/releases_and_maturity/migration/migration_guide_2.8.rst]: Agreed, I have elaborated with an example, is that fine? https://github.com/nrfconnect/sdk-nrf/pull/16753/files#diff-73508775c4364c7221f61e1b098cd77eaf956e37eb209b707c649bc2b9a987c7.

Comment (trantanen) [reply doc/nrf/releases_and_maturity/migration/migration_guide_2.8.rst]: Looks good to me!

**Review by trantanen — APPROVED** [review]

**Review by richabp — APPROVED** [review]

**Review by sachinthegreen — APPROVED** [review]

**Comment by shanthanordic** [comment]
This should not be merged by commeting test case without @endre-nordic  approval

**Comment by krish2718** [comment]
> This should not be merged by commeting test case without @endre-nordic approval

Sure,  after discussing with @bjarki-andreasen it was concluded to cherry-pick fixes from this PR to the upmerge PR and we can review it there.

File: samples/cellular/nrf_cloud_multi_service/sample.yaml
Comment (glarsennordic) [inline samples/cellular/nrf_cloud_multi_service/sample.yaml:97]: This is a connectivity build, not a scan-only build, so I think that means we should stick with `nrf7002ek`, right?

File: samples/cellular/nrf_cloud_multi_service/sample.yaml
Comment (glarsennordic) [inline samples/cellular/nrf_cloud_multi_service/sample.yaml:87]: There are some references to this shield in `` that I think also need to be updated. They are all in the `Building with nRF7002 Wi-Fi scanning support` section.

Specifically, lines 797, 804. These should, if I am not mistaken, switch to using the `nrf7002ek` shield.

But beware, any mentions of the shield outside this section are for connectivity-enabled builds, so I don't think those need to change.

**Review by glarsennordic — CHANGES_REQUESTED** [review]

**Comment by glarsennordic** [comment]
Any updates regarding this / regarding the upmerge?

**Comment by krish2718** [comment]
All changes in this PR are part of upmerge, any missing changes will be take care for 2.8.0.

**Pull request closed without merging**
